src/libraries/System.Private.CoreLib/gen/EventSourceGenerator.Parser.cs

```diff
@@ -4,119 +4,96 @@
-                            continue;
-                        }
-                    }
-                    else
+                    switch (argName)
                     {
-                        nspace = ns.Name.ToString();
-                        while (true)
-                        {
-                            ns = ns.Parent as NamespaceDeclarationSyntax;
-                            if (ns == null)
-                            {
-                                break;
-                            }
-
-                            nspace = $"{ns.Name}.{nspace}";
-                        }
+                        case "Guid":
+                            guid = value;
+                            break;
+                        case "Name":
+                            name = value;
+                            break;
                     }
+                }
 
-                    string className = classDef.Identifier.ToString();
-                    string name = className;
-                    string guid = "";
+                if (!Guid.TryParse(guid, out Guid result))
+                {
+                    result = GenerateGuidFromName(name.ToUpperInvariant());
+                }
 
-                    SeparatedSyntaxList<AttributeArgumentSyntax>? args = ca.ArgumentList?.Arguments;
-                    if (args is not null)
-                    {
-                        foreach (AttributeArgumentSyntax arg in args)
-                        {
-                            string argName = arg.NameEquals!.Name.Identifier.ToString();
-                            string value = sm.GetConstantValue(arg.Expression, cancellationToken).ToString();
-
-                            switch (argName)
-                            {
-                                case "Guid":
-                                    guid = value;
-                                    break;
-                                case "Name":
-                                    name = value;
-                                    break;
-                            }
-                        }
-                    }
+                eventSourceClass = new EventSourceClass(nspace, className, name, result);
+                continue;
+            }
 
-                    if (!Guid.TryParse(guid, out Guid result))
-                    {
-                        result = GenerateGuidFromName(name.ToUpperInvariant());
-                    }
+            return eventSourceClass;
+        }
 
-                    eventSourceClass = new EventSourceClass(nspace, className, name, result);
-                    continue;
-                }
-            }
+        private static string? ConstructNamespace(NamespaceDeclarationSyntax? ns)
+        {
+            if (ns is null)
+                return string.Empty;
 
-            if (!autoGenerate)
+            string nspace = ns.Name.ToString();
```

Comment:
note.  this is not correct.  if you have `namespace N1/*comment*/. N2`, then the ToString will include hte extraneous comments and whitespace.